ironic/workflow.go

```diff
@@ -285,10 +285,11 @@ func (workflow *provisionStateWorkflow) buildProvisionStateOpts(target nodes.Tar
 		}
 	}
 	if target == "clean" {
-		opts.CleanSteps = []nodes.CleanStep{}
-		// TODO if we want to actually clean, then we need clean_steps
-		// currently bmo does quite a lot of work to get raid cleaning working.
-		// https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L1249-L1292
+		if workflow.configDrive != nil {
+			opts.CleanSteps = workflow.configDrive.([]nodes.CleanStep)
```

Comment:
Shouldn't this be `workflow.cleanSteps`, similarly to line 284? I don't quite get how `configDrive` is related to cleaning.

Reply:
Yes, I have added a new filed `cleanStep` to the struct  `provisionStateWorkflow`. PTAL https://github.com/openshift-metal3/terraform-provider-ironic/pull/53/commits/8a5b0d38726f8730df8d42e9dbbdf491d1d847e8